docs/proposals/proposal-002-run.md

```diff
@@ -0,0 +1,312 @@
+* What a test workflow must contain in order to run on the cluster
+* How a test workflow is related to test instructions
+
+At a bare minimum, the test workflow must contain test instructions of what should run in the Kubernetes cluster. For example, the Falco project maintainers have identified that one way to test the Falco project is through a test that runs `stress-ng` for a given period of time. The test instructions are contained in a Deployment manifest which can be directly applied to the community cluster using `kubectl`
+
+The test workflows will be stored in the same JSON file as the other parameters for CNCF projects as defined in [Proposal 1](./proposal-001-trigger-and-deploy.md). It can be added as an additional input.
+
+```yaml
+# .github/workflows/benchmark-pipeline.yaml
+jobs:
+  # first, must authenticate to the Kubernetes cluster
+  # this is a Job
+  test-job:
+    # test job calls on test workflow
+    uses: ${{ inputs.test_path }} # refers to test workflow path
+```
+
+This will fetch the workflow using the `jobs.<job_id>.uses` syntax defined [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iduses).
+
+Below are two use cases: the test workflow may be defined in the Green Reviews repository or in a separate repository.
+
+#### Use Case 1: A GitHub Action job using a workflow defined in the _same_ repository (preferred)
+
+If the test workflow is located in the Green Reviews repository, `test_path` would refer to, for example, `cncf-tags/green-reviews-tooling/.github/workflows/falco-test-workflow.yml@v1`.
+
+In terms of the directory structure, in the `green-reviews-tooling` repository, we could create a subfolder such as `./github/workflows/tests` to contain the test workflows.
+
+#### Use Case 2: A GitHub Action job using a workflow defined in a _different_ repository
+
+We want to accommodate different methods of setting up the tests depending on the CNCF project. Given this, the test workflow containing the test instructions could be defined in a different repository. In this case, the `test_path` would be, for example, `falcosecurity/cncf-green-review-testing/.github/workflows/workflow.yml@v1`.
+
+![Pipeline run](diagrams/pipeline-run.png "An example pipeline run")
+
+### Test instructions
+
+The test workflow which contains the test instructions may look like the following:
+
+```yaml
+# .github/workflows/tests/falco-test-workflow.yaml
+jobs:
+  stress-ng-test:
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          # the action to take here depends on the Functional Unit of the CNCF project. wait for amount of time, for resources
+          kubectl apply -f https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/main/kustomize/falco-driver/ebpf/stress-ng.yaml
+          wait 15m
+          kubectl delete -f https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/main/kustomize/falco-driver/ebpf/stress-ng.yaml
```

Comment:
Something like this?
```suggestion
      - delete: |
         kubectl delete -f https://raw.githubusercontent.com/falcosecurity/cncf-green-review-testing/main/kustomize/falco-driver/ebpf/stress-ng.yaml
```

Reply:
I put it in a different workflow, as I understood that was the suggestion of @rossf7. Take a look after I push the changes and let me know if it's ok

Reply:
@locomundo sorry my feedback is late here :(

I think it should be a separate job like @AntonioDiTuri suggests with `if: ${{ always() }}` like we do here. So whatever happens the cleanup logic is executed. 

https://github.com/rossf7/green-reviews-tooling/blob/73bc1fd557c09d052aebab0b52613ae615ed0490/.github/workflows/benchmark-pipeline.yaml#L65

Calling a separate workflow is tricky because cron workflow would need to wait until the project workflow completes.

Reply:
check, changing it!

Reply:
@locomundo thank you! That looks good 👍